### PR TITLE
Add retry for lambda update conflict

### PR DESCRIPTION
**Pull request by julienduchesne**
Fixes https://github.com/terraform-providers/terraform-provider-aws/issues/5154

The issue was fixed on the lambda_permission resource's side but not on the lambda_function side. So if you only update the permissions, you're fine but if you modify the lambda itself as well as the permissions, it crashes

### CODE REVIEW

File: aws/resource_aws_lambda_function.go [hunk 1]
```diff
@@ -539,6 +539,11 @@ func resourceAwsLambdaFunctionCreate(d *schema.ResourceData, meta interface{}) e
 			return resource.RetryableError(err)
 		}
 
+		if tfawserr.ErrMessageContains(err, lambda.ErrCodeResourceConflictException, "") {
```
Comment (ewbankkit) [inline aws/resource_aws_lambda_function.go:542]: Simpler.
```suggestion
		if tfawserr.ErrCodeEquals(err, lambda.ErrCodeResourceConflictException) {
```

File: aws/resource_aws_lambda_function.go [hunk 2]
```diff
@@ -1060,6 +1065,11 @@ func resourceAwsLambdaFunctionUpdate(d *schema.ResourceData, meta interface{}) e
 				return resource.RetryableError(err)
 			}
 
+			if tfawserr.ErrMessageContains(err, lambda.ErrCodeResourceConflictException, "") {
```
Comment (ewbankkit) [inline aws/resource_aws_lambda_function.go:1068]: Simpler.
```suggestion
			if tfawserr.ErrCodeEquals(err, lambda.ErrCodeResourceConflictException) {
```

**Comment by ewbankkit** [comment]
This same error code is handled similarly in `aws/resource_aws_lambda_event_source_mapping.go`.

**Comment by anGie44** [comment]
Hi @julienduchesne , thank you for raising this PR. Changes  LGTM, only thing is we need to rebase this PR and unfortunately i do not have read/write access to the remote repo this PR was made from. To get this merged, I'm going to close this one and open a new PR with these changes.

**Pull request closed without merging**